SensorApplication.java

```diff
@@ -0,0 +1,54 @@
+package sensor;
+
+import java.awt.Dimension;
+import java.awt.GridLayout;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.border.TitledBorder;
+
+public class SensorApplication extends JFrame {
+	
+	public SensorApplication() {
+		setTitle("Sensor Tracker");
+		setLayout(new GridLayout(3,1));
+	
+		TemperatureSensor ts = new TemperatureSensor();
+		SensorAdapter ta = new TemperatureSensorAdapter(ts);
+		JPanel  temperaturePnl = addPanel(ta);
+		add(temperaturePnl);
+		
+		PressureSensor ps = new PressureSensor();
+		SensorAdapter pa = new PressureSensorAdapter(ps);
+		JPanel  pressurePnl = addPanel(pa);
+		add(pressurePnl);
+		
+		RadiationSensor rs = new RadiationSensor();
+		SensorAdapter ra = new RadiationSensorAdapter(rs);
+		JPanel  radiationPnl = addPanel(ra);
+		add(radiationPnl);	
+		
+		setPreferredSize(new Dimension(600,600));
+		setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+		setVisible(true);
+		pack();
+	}
+	
+	public JPanel addPanel(SensorAdapter sa) {

```

Comment:
Nice thinking using a function that takes the sensor adapter as a parameter in order to build the GUI - adds modularity